### ISSUE TITLE
Can't read array if dimension has unsigned domain but signed dtype

### ISSUE THREAD

**Issue by gatesn** [TileDB-Inc/TileDB-Py#572]
It looks like the signed-ness of the dtype is inferred from the domain, rather than declared?

tiledb-py 0.8.2
tiledb 2.2.3

```python
from tiledb import *
import numpy as np

array_name = 'unsigned'
s = ArraySchema(domain=Domain(Dim('a', dtype=np.int8, domain=(0, 126))), attrs=[Attr('x', dtype=np.int32)], sparse=True)

A = SparseArray.create(array_name, schema=s)
with SparseArray(array_name, mode='w') as A:
    A[0] = 1

with SparseArray(array_name, mode='r') as A:
    print(A[:])
```

```
Traceback (most recent call last):
  File "XXXX", line 12, in <module>
    print(A[:])
  File "tiledb/libtiledb.pyx", line 5237, in tiledb.libtiledb.SparseArrayImpl.__getitem__
  File "tiledb/libtiledb.pyx", line 5374, in tiledb.libtiledb.SparseArrayImpl.subarray
  File "tiledb/libtiledb.pyx", line 5392, in tiledb.libtiledb.SparseArrayImpl._read_sparse_subarray
tiledb.libtiledb.TileDBError: Static type (UINT8) does not match expected type (INT8)
```

**Comment by ihnorton** [comment]
Hi @gatesn, thanks for opening this issue. We've fixed it in #573, which will be included in the next patch release, likely Monday next week.

**Comment by ihnorton** [comment]
(note: [build from source](https://docs.tiledb.com/main/solutions/tiledb-embedded/installation/building-from-source/python) in the interim, in case needed)

**Comment by ihnorton** [comment]
Circling back to this; I've confirmed the fix in TileDB-Py 0.8.11:

```
$ python py572.py
OrderedDict([('x', array([1], dtype=int32)), ('a', array([0], dtype=int8))])
```

```
python -c 'import tiledb; print(tiledb.version.version)'
0.8.11
```

**Comment by gatesn** [comment]
I can also confirm fixed on my end with 0.8.11, thank you!
